xbmc/addons/AddonDll.cpp

```diff
@@ -461,5 +398,31 @@ ADDON_STATUS CAddonDll::TransferSettings()
   return ADDON_STATUS_OK;
 }
 
+bool CAddonDll::CheckAPIVersion(int type)
+{
+  /* check the API version */
+  const char* kodiVersion = kodi::addon::GetTypeVersion(type);
+  const char* addonVersion = m_pDll->GetAddonTypeVersion(type);
+
+  if (AddonVersion(addonVersion) != AddonVersion(kodiVersion))
+  {
+    CLog::Log(LOGERROR, "Add-on '%s' is using an incompatible API version for type '%s'. Kodi API version = '%s', add-on API version '%s'",
+                            Name().c_str(),
+                            kodi::addon::GetTypeName(type),
+                            kodiVersion,
+                            addonVersion);
+
+    std::string heading = StringUtils::Format("%s: %s", TranslateType(Type(), true).c_str(), Name().c_str());
+    std::string text = StringUtils::Format(g_localizeStrings.Get(29803).c_str(),
+                                           kodi::addon::GetTypeName(type),
+                                           kodiVersion,
+                                           addonVersion);
+    CGUIDialogOK::ShowAndGetInput(CVariant{heading}, CVariant{text});
```

Comment:
Have no idea for really good alternatives, want to have them only be available there, that's why I find a user message as not so good way. As alternative can I use this:
```cpp
CEventLog::GetInstance().AddWithNotification(EventPtr(new CNotificationEvent(heading, text, EventLevel::Error)));
```
who bring:
![bildschirmfoto vom 2017-04-27 02-04-08](https://cloud.githubusercontent.com/assets/6879739/25462453/609cca4e-2aef-11e7-9b29-e052f6104ecc.png)

Here how it was with them here:
![bildschirmfoto vom 2017-04-27 02-09-31](https://cloud.githubusercontent.com/assets/6879739/25462460/75b302cc-2aef-11e7-956a-17e4f0a05e69.png)